pkg/catalog/outbound_traffic_policies.go

```diff
@@ -106,13 +106,16 @@ func (mc *MeshCatalog) listOutboundTrafficPoliciesForTrafficSplits(sourceNamespa
 	return outboundPoliciesFromSplits
 }
 
+// ListOutboundServicesForGateway lists the upstream services for the gateway
+func (mc *MeshCatalog) ListOutboundServicesForGateway() []service.MeshService {
```

Comment:
Instead of a new API, is it not possible to use `ListOutboundServicesForIdentity`, and match the `ServiceIdenity` to something osm-controller is aware of, ex. osm-gateway.osm-system.cluster.local? We could create a method `ServiceIdentity.IsGateway()` for this.

Reply:
As discussed offline, creating a new API will be cleaner for now as implementing `ServiceIdentity.IsGateway()` will need the osm sa which is the release name and can vary. We could improve this in the future my hardcoding the service account for the gateway.